File: algds/codewars/6kyu/multiples-of-3-or-5/c/sum_mults_3_or_5_v1.c
//
// tags: javascript codewars 6kyu math algorithm multiple
//

#include <stdio.h>

/**
 * Checks whether `y` is a multiple of `x`.
 */
int is_mult_of(int x, int y) {
  return y % x == 0;
}

int sum_mults_3_or_5(int n) {
  int total = 0;

  for (int i = 1; i < n; ++i)
    if (is_mult_of(5, i) || is_mult_of(3, i))
      total += i;

  return total;
}

int main(void) {

  printf("%d\n", sum_mults_3_or_5(10));

  return 0;
}

File: codewars/6kyu/sum-multiples-of-3-or-5/js/sumMults3or5_v1.js
//
// tags: javascript codewars 6kyu math algorithm multiple
//

const log = console.log.bind(console);

/**
 * Checks whether `y` is a multiple of `x`.
 *
 * @param {number} x
 * @param {number} y
 * @returns {boolean}
 */
function isMultOf(x, y) {
  return y % x === 0;
}

function sumMults3or5(n) {
  if (n < 0) return 0;

  let total = 0;

  for (let i = 1; i < n; ++i) {
    if (isMultOf(3, i) && isMultOf(5, i))
      total += i;
    else if (isMultOf(3, i))
      total += i;
    else if (isMultOf(5, i))
      total += i;
  }

  return total;
}

log(sumMults3or5(10));
